### PR TITLE
BUG: Series.__delitem__ converting EAs to ndarrays

### LINKED ISSUE #40386: BUG: `Series.__delete__` with EA casts to object
it calls np.delete, which casts to ndarray, which for e.g. PeriodDtype will cast to object

```
dti = pd.date_range("2016-01-01", periods=3, tz="US/Pacific")
ser = pd.Series(dti)

del ser[1]

>>> ser
0    2016-01-01 00:00:00-08:00
2    2016-01-03 00:00:00-08:00
dtype: object
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #40386
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -1338,7 +1338,28 @@ def quantile(
         return new_block(result, placement=self._mgr_locs, ndim=2)
 
 
-class ExtensionBlock(Block):
+class EABackedBlock(Block):
```
Comment (jreback) [inline pandas/core/internals/blocks.py:1341]: 2 bad you have to do this (create a nother block type), any better way (e.g. handle it in the EA impl)?

Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:1341]: there are actually a few other methods im planning to move here to share between ExtensionBlock/NDArrayBackedExtensionBlock.  I think the code de-duplication makes this a net plus

File: pandas/tests/series/indexing/test_delitem.py [hunk 2]
```diff
@@ -50,3 +51,23 @@ def test_delitem_missing_key(self):
 
         with pytest.raises(KeyError, match=r"^0$"):
             del s[0]
+
+    def test_delitem_extension_dtype(self):
```
Comment (jreback) [inline pandas/tests/series/indexing/test_delitem.py:55]: maybe worthile adding all EA types here?

Comment (jbrockmendel) [reply pandas/tests/series/indexing/test_delitem.py:55]: ill add a test in the extension tests next pass

**Pull request merged**